### DISCUSSION TITLE
`yarn dev` type checking

### DISCUSSION THREAD

**Discussion by starpit** [alex8088/electron-vite#310] [Q&A]
Howdy, thanks so much for the great work. We are trying to figure out how to enable type checking while in dev/watch mode. Carrying this over from our non-electron vite base, we added `vite-plugin-checker` to our electron.vite.config.js. But it does not seem to catch any typescript errors, neither in main nor render code.

**Comment by alex8088** [comment]
you need to set `tsconfigPath` for checker

**Reply by onesianinflaredguard** [reply]
Alex do you stay in island

**Comment by starpit** [comment]
Thanks. I think the issue is that the starter kit we used split the tsconfig into a web and node version, and the main tsconfig uses `references` to point to those two separate configs. Apparently vite-plugin-checker doesn't work with a tsconfig-with-references. When I update the vite config to point to one of them, then the type checking kicks in. Hmm..

Update: `buildMode: true` is what we need. This tells vite-plugin-checker to run in `tsc --build` mode, which is what is needed to support composite tsconfigs.
